File: QuantLib-SWIG/Python/test/marketelements.py
"""
 Copyright (C) 2000, 2001, 2002, 2003 RiskMap srl

 This file is part of QuantLib, a free-software/open-source library
 for financial quantitative analysts and developers - http://quantlib.org/

 QuantLib is free software: you can redistribute it and/or modify it under the
 terms of the QuantLib license.  You should have received a copy of the
 license along with this program; if not, please email quantlib-dev@lists.sf.net
 The license is also available online at http://quantlib.org/html/license.html

 This program is distributed in the hope that it will be useful, but WITHOUT
 ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 FOR A PARTICULAR PURPOSE.  See the license for more details.
"""

import QuantLib
import unittest

flag = None
def raiseFlag():
    global flag
    flag = 1

class MarketElementTest(unittest.TestCase):
    def testObservable(self):
        "Testing observability of market elements"
        global flag
        flag = None
        me = QuantLib.SimpleQuote(0.0)
        obs = QuantLib.Observer(raiseFlag)
        obs.registerWith(me)
        me.setValue(3.14)
        if not flag:
            self.fail("Observer was not notified of market element change")
    def testObservableHandle(self):
        "Testing observability of market element handles"
        global flag
        flag = None
        me1 = QuantLib.SimpleQuote(0.0)
        h = QuantLib.QuoteHandle(me1)
        obs = QuantLib.Observer(raiseFlag)
        obs.registerWith(h)
        me1.setValue(3.14)
        if not flag:
            self.fail("Observer was not notified of market element change")
        flag = None
        me2 = QuantLib.SimpleQuote(0.0)
        h.linkTo(me2)
        if not flag:
            self.fail("Observer was not notified of market element change")


if __name__ == '__main__':
    print 'testing QuantLib', QuantLib.__version__
    suite = unittest.TestSuite()
    suite.addTest(unittest.makeSuite(MarketElementTest,'test'))
    unittest.TextTestRunner(verbosity=2).run(suite)
    raw_input('press return to continue')

File: QuantLib-SWIG/Ruby/test/marketelements.rb
=begin
 Copyright (C) 2000, 2001, 2002 RiskMap srl

 This file is part of QuantLib, a free-software/open-source library
 for financial quantitative analysts and developers - http://quantlib.org/

 QuantLib is free software: you can redistribute it and/or modify it under the
 terms of the QuantLib license.  You should have received a copy of the
 license along with this program; if not, please email ferdinando@ametrano.net
 The license is also available online at http://quantlib.org/html/license.html

 This program is distributed in the hope that it will be useful, but WITHOUT
 ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 FOR A PARTICULAR PURPOSE.  See the license for more details.
=end

# $Id$

require 'QuantLib'
require 'runit/testcase'
require 'runit/cui/testrunner'

class MarketElementTest < RUNIT::TestCase
  def name
    "Testing observability of market elements..."
  end
  def test
    flag = false
    me = QuantLib::SimpleMarketElement.new(0.0)
    obs = QuantLib::Observer.new { flag = true }
    obs.registerWith(me)
    me.value = 3.14
    unless flag
        assert_fail("Observer was not notified of market element change")
    end
  end
end

class MarketElementHandleTest < RUNIT::TestCase
  def name
    "Testing observability of market element handles..."
  end
  def test
    flag = false
    me1 = QuantLib::SimpleMarketElement.new(0.0)
    h = QuantLib::MarketElementHandle.new(me1)
    obs = QuantLib::Observer.new { flag = true }
    obs.registerWith(h)
    me1.value = 3.14
    unless flag
        assert_fail("Observer was not notified of market element change")
    end
    flag = false
    me2 = QuantLib::SimpleMarketElement.new(0.0)
    h.linkTo!(me2)
    unless flag
        assert_fail("Observer was not notified of market element change")
    end
  end
end


if $0 == __FILE__
  RUNIT::CUI::TestRunner.run(MarketElementTest.suite)
  RUNIT::CUI::TestRunner.run(MarketElementHandleTest.suite)
end

